lib/eunit/src/eunit.erl

```diff
@@ -265,3 +266,49 @@ all_options(Opts) ->
     catch
 	_:_ -> Opts
     end.
+
+%% Delete duplicates of tests
+
+delete_duplicates([_|_] = Tests) ->
+    Candidates = [atom_to_list(T) || T <- Tests, is_atom(T), is_mod_exist(T)],
+    delete_duplicates(Candidates, Tests);
+delete_duplicates(Tests) ->
+    Tests.
+
+is_mod_exist(M) ->
+    try M:module_info(exports) of
+    _ ->
+        true
+    catch
+    _:_ ->
+        false
+    end.
+
+delete_duplicates(Candidates, Tests) ->
+    delete_duplicates(Candidates, Candidates, Tests).
+
+delete_duplicates([], _, Tests) ->
+    Tests;
+delete_duplicates([H|T], Candidates, Tests) ->
+    delete_duplicates(T, Candidates, delete_candidate(H, Candidates, Tests)).
+
+delete_candidate(Name, Candidates, Tests) ->
+    case lists:suffix(?DEFAULT_TESTMODULE_SUFFIX, Name) of
+        true ->
+            Delta = length(Name) - length(?DEFAULT_TESTMODULE_SUFFIX),
+            CatName = lists:sublist(Name, 1, Delta),
+            case lists:member(CatName, Candidates) of
+                true ->
+                    lists:delete(list_to_atom(Name), Tests);
+                false ->
+                    Tests
+            end;
+        false ->
+            Candidate = list_to_atom(Name ++ ?DEFAULT_TESTMODULE_SUFFIX),
+            case lists:member(Candidate, Tests) of
+                true ->
+                    lists:delete(Candidate, Tests);
+                false ->
+                    Tests
+            end
```

Comment:
It seems unnecessary to test for membership before deleting since the `Tests` lists unlikely to be very long.

```suggestion
            Tests -- [Candidate]
```

Reply:
Done.